markdown/extensions/meta.py

```diff
@@ -36,10 +36,18 @@ class MetaExtension (Extension):
 
     def extendMarkdown(self, md, md_globals):
         """ Add MetaPreprocessor to Markdown instance. """
+        md.registerExtension(self)
+        self.md = md
         md.preprocessors.add("meta",
                              MetaPreprocessor(md),
                              ">normalize_whitespace")
 
+    def reset(self):
+        try:
+            del self.md.Meta
```

Comment:
No need to delete it with a try/except. Just set it to `self.md.Meta = {}`.

Reply:
That's a good remark. However, there would be a discrepancy between the initial state (when there is no `Meta`) and after a call to `reset()`, even if there is no call to `convert()` first.

I am not used to this library, but I'd expect a `reset()` method to remove every useless parts that are inexpensive to rebuild.

Reply:
Fair enough.  It looks like the resets don't get called when the class is initialized.  We could just set `md.Meta = {}` in `extendMarkdown` and have a consistent interface that doesn't require deleting the `Meta` attribute.  I think that would be my personal preference, but I'll let @waylan make the final decision.

Reply:
Never mind, reset does get called on initialization.  I personally think it makes sense to initialize `Meta` as `{}`.  Then you don't have to remember if you set reset or not.  It's filled when it has something important, and empty when it doesn't (reset being one of these cases). waylan will let you know what he prefers, but that is my preference.  I think meta should initialize all of the variables it is going to use (in this case `Meta`).

Reply:
I agree with @facelessuser. Set `Meta` at `{}` in `reset`. `reset` gets called when the instance is first created, so that will ensure that the attribute always exists and always contains a dict.

Reply:
Done in 7f8ce8a